lib/taurus/qt/qtgui/taurusgui/config_loader/abstract.py

```diff
@@ -0,0 +1,180 @@
+
+
+__all__ = ["AbstractConfigLoader", "ConfigLoaderError"]
+
+
+class ConfigLoaderError(Exception):
+    """
+    Base exception raised by ConfigLoader
+    """
+
+    def __init__(self, message):
+        message = "Exception raised while loading configuration: " + message
+        super(ConfigLoaderError, self).__init__(message)
+
+
+class AbstractConfigLoader(abc.ABC):
+    """
+    Abstract class for config loaders.
+    It defines interface which has to be implemneted by subclass in
+    order for it to be ConfigLoader
+    """
+
+    def __init__(self, confname):
+        self._confname = confname
+
+    @abc.abstractmethod
+    def supports(self, confname):
+        """
+        Return True or False for support of specific configuration passed
+        """
+        return None
+
+    @abc.abstractmethod
+    def load(self):
+        """
+        This method is meant to load actual data from file on disk
+        """
+
+    @abc.abstractproperty
+    def conf_dir(self):
+        """
+        Absolute path to directory in which config file is placed
+        """
+        return None
+
+    @abc.abstractproperty
+    def app_name(self):
+        """Name of the application"""
+        return None
+
+    @abc.abstractproperty
+    def org_name(self):
+        """Name of organization"""
+        return None
+
+    @abc.abstractproperty
+    def custom_logo(self):
+        """Path to application's custom logo file"""
+        return None
+
+    @abc.abstractproperty
+    def org_logo(self):
+        """Path to organization's custom logo file"""
+        return None
+
+    @abc.abstractproperty
+    def single_instance(self):
+        """Whether more than one instance of application can be launched simultaously or not"""
+        return None
+
+    @abc.abstractproperty
+    def manual_uri(self):
+        """URI pointing to application's manual"""
+        return None
+
+    @abc.abstractproperty
+    def ini_file(self):
+        """Path to application's default INI file with settings"""
+        return None
+
+    @abc.abstractproperty
+    def extra_catalog_widgets(self):
+        """Path to application's custom logo file"""
+        return []
+
+    @abc.abstractproperty
+    def synoptics(self):
+        """Sequence of paths to synoptic files"""
+        return []
+
+    @abc.abstractproperty
+    def console(self):
+        """Whether to add console widget or not"""
+        return None
+
+    @abc.abstractproperty
+    def panels(self):
+        """List of custom panels with widgets"""
+        return []
+
+    @abc.abstractproperty
+    def toolbars(self):
+        """List of custom toolbars"""
+        return []
+
+    @abc.abstractproperty
+    def applets(self):
+        """List of custom applets"""
+        return []
+
+    @abc.abstractproperty
+    def external_apps(self):
+        """List of external applications"""
+        return []
+
+    # deprecated
+    @abc.abstractproperty
+    def monitor(self):
```

Comment:
As already discussed, this property (and maybe other deprecated ones) could be left out of the mandatory API. In fact, since we allow multiple loaders, they could be moved to a "bck-compat" type of loader

Reply:
Will be moved to `bck-compat` config loader. What about Sardana options? New loader like `sardana.something`?